[NFC][clang-tidy] Add type annotations to check_clang_tidy

nicovank opened the pull request:
I'm looking to make some changes in this file, start by typing it.

```
> python3 -m mypy --strict clang-tools-extra/test/clang-tidy/check_clang_tidy.py
Success: no issues found in 1 source file
```

EugeneZelenko commented on clang-tools-extra/test/clang-tidy/check_clang_tidy.py:
```diff
@@ -71,12 +72,12 @@ def try_run(args, raise_error=True):
 # This class represents the appearance of a message prefix in a file.
 class MessagePrefix:
-    def __init__(self, label):
+    def __init__(self, label: str):
```
```suggestion
    def __init__(self, label: str) -> None:
```

nicovank replied:
Good catch, thank you, I always forget those because MyPy doesn't flag it.

nicovank commented on clang-tools-extra/test/clang-tidy/check_clang_tidy.py:
```diff
@@ -48,15 +48,16 @@
 import re
 import subprocess
 import sys
+from typing import List, Tuple
```
`list` and `tuple` are Python 3.9: https://peps.python.org/pep-0585/
LLVM minimum Python version is 3.8 for now: https://github.com/llvm/llvm-project/blob/82c078c54d16b20c65d5e361f5da58b0b83c8c47/llvm/CMakeLists.txt#L973

Although 3.8 is EOL, I wouldn't be surprised there are machines somewhere still running the test suites on 3.8.

nicovank commented on clang-tools-extra/test/clang-tidy/check_clang_tidy.py:
```diff
@@ -71,12 +72,13 @@ def try_run(args, raise_error=True):
 # This class represents the appearance of a message prefix in a file.
 class MessagePrefix:
-    def __init__(self, label):
+
```
Note that this space is "on purpose", the formatter wants it there.

nicovank committed:
[NFC][clang-tidy] Add type annotations to check_clang_tidy

I'm looking to make some changes in this file, start by typing it.

```
> python3 -m mypy --strict clang-tools-extra/test/clang-tidy/check_clang_tidy.py
Success: no issues found in 1 source file
```
